<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>&lt;no title&gt; &mdash; YeoLab-welcome 0.1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/basic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/bootswatch-3.2.0/lumen/bootstrap.min.css" type="text/css" />
    <link rel="stylesheet" href="_static/bootstrap-sphinx.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="_static/bootstrap-3.2.0/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="_static/bootstrap-sphinx.js"></script>
    <link rel="top" title="YeoLab-welcome 0.1.0 documentation" href="index.html" />
<meta charset='utf-8'>
<meta http-equiv='X-UA-Compatible' content='IE=edge,chrome=1'>
<meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1'>
<meta name="apple-mobile-web-app-capable" content="yes">

  </head>
  <body role="document">

  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="index.html">
          Yeolab Onboarding</a>
        <span class="navbar-text navbar-version pull-left"><b>0.1.0</b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
                <li><a href="http://yeolab.ucsd.edu/yeolab/Home.html">YeoLab@UCSD</a></li>
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="index.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul>
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting Started</a><ul>
<li class="toctree-l2"><a class="reference internal" href="getting_started.html#sanford-internet-access">Sanford Internet Access</a></li>
<li class="toctree-l2"><a class="reference internal" href="getting_started.html#printing">Printing</a></li>
<li class="toctree-l2"><a class="reference internal" href="getting_started.html#reserving-rooms-in-scrm-building">Reserving rooms in SCRM building</a></li>
<li class="toctree-l2"><a class="reference internal" href="getting_started.html#access-to-compute-environments">Access to Compute Environments</a></li>
<li class="toctree-l2"><a class="reference internal" href="getting_started.html#online-resources">Online Resources</a></li>
<li class="toctree-l2"><a class="reference internal" href="getting_started.html#internal-resources">Internal Resources</a></li>
</ul>
</li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="tscc.html">TSCC</a><ul>
<li class="toctree-l2"><a class="reference internal" href="tscc.html#first-steps">First Steps</a></li>
<li class="toctree-l2"><a class="reference internal" href="tscc.html#installing-and-upgrading-python-packages">Installing and upgrading Python packages</a></li>
<li class="toctree-l2"><a class="reference internal" href="tscc.html#submitting-and-managing-compute-jobs-on-tscc">Submitting and managing compute jobs on TSCC</a></li>
<li class="toctree-l2"><a class="reference internal" href="tscc.html#ipython-notebooks-on-tscc">IPython notebooks on TSCC</a></li>
<li class="toctree-l2"><a class="reference internal" href="tscc.html#running-qscripts-gatk-queue-pipelines-on-tscc">Running qscripts GATK Queue pipelines on TSCC</a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul class="simple">
</ul>
</ul>
</li>
              
            
            
              
                
              
            
            
            
            
              <li class="hidden-sm">
<div id="sourcelink">
  <a href="_sources/analyze_rna_seq.txt"
     rel="nofollow">Source</a>
</div></li>
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">
    <div class="col-md-12">
      
  <p>The file extension below are generated by RNA-seq pipeline, the prefix for each file is the name fed in by the manifest</p>
<dl class="docutils">
<dt>polyATrim.adapterTrim.fastq.out</dt>
<dd>Std out for cutadapt</dd>
<dt>polyATrim.adapterTrim.metrics</dt>
<dd>cutadapt metrics</dd>
<dt>polyATrim.adapterTrim.rmRep_fastqc.zip</dt>
<dd>zip file for fastqc  post-cutadapt, post, repetitive element removal</dd>
<dt>polyATrim.adapterTrim.rmRep.fastq.out</dt>
<dd>std out for repetitive element removal</dd>
<dt>polyATrim.adapterTrim.rmRep.metrics</dt>
<dd>metrics for number of repetitive elements removed</dd>
<dt>polyATrim.adapterTrim.rmRep.sam</dt>
<dd>sam file mapped to reference genome</dd>
<dt>polyATrim.adapterTrim.rmRep.samLog.final.out</dt>
<dd>STAR specific output</dd>
<dt>polyATrim.adapterTrim.rmRep.samLog.out</dt>
<dd>STAR Specific output</dd>
<dt>polyATrim.adapterTrim.rmRep.samLog.progress.out</dt>
<dd>STAR Specific output</dd>
<dt>polyATrim.adapterTrim.rmRep.samLog.std.out</dt>
<dd>STAR Specific output</dd>
<dt>polyATrim.adapterTrim.rmRep.sam.out</dt>
<dd>STAR mapping std out</dd>
<dt>polyATrim.adapterTrim.rmRep.samSJ.out.tab</dt>
<dd>STAR specific output</dd>
<dt>polyATrim.adapterTrim.rmRep.samUnmapped.out.mate1</dt>
<dd>STAR specific output</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.bai</dt>
<dd>index file for intermediate processing, ignore</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.bam</dt>
<dd>intermediate bam file, ignore</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.bam.out</dt>
<dd>std out for STAR, ignore</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.bai</dt>
<dd>index file for intermediate processing, ignore</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.bam</dt>
<dd>final read, group assigned bam file</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.bam.bai</dt>
<dd>final read, group assigned bam index file</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.bam.bai.out</dt>
<dd>stdout for final read, group assigned bam index file</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.bam.out</dt>
<dd>stdout for final read, group assigned bam file</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.count</dt>
<dd>Counts of reads at in each gene, from CountTags</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.count.out</dt>
<dd>stdout for Counts of reads at in each gene, from CountTags</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.editing.out</dt>
<dd>editing make file out</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.make.out</dt>
<dd>unknown ignore</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.miso.out</dt>
<dd>miso make file out</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.bg.out</dt>
<dd>stdout for genomeCoverageBed, negative strand</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.bw</dt>
<dd>bigWig file showing read coverage along the negative strand, transposed so values are negative</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.bw.out</dt>
<dd>stdout for bw file generation</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.norm.bg.out</dt>
<dd>stdout for bg file generation</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.norm.bw</dt>
<dd>read depth normalized, transposed negative strand coverage bigWig file</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.norm.bw.out</dt>
<dd>stdout for bigwig file generation</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.NRF.metrics</dt>
<dd>metrics for PCR duplication, for reference see: <a class="reference external" href="http://genome.ucsc.edu/ENCODE/qualityMetrics.html">http://genome.ucsc.edu/ENCODE/qualityMetrics.html</a></dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.NRF.metrics.out</dt>
<dd>stdout for NRF calculations</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.pos.bg.out</dt>
<dd>stdout for bedgraph generation</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.pos.bw</dt>
<dd>bigWig file showing coverage along positive strand</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.pos.bw.out</dt>
<dd>stdout for bigwig generation</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.pos.norm.bg.out</dt>
<dd>stdout for bedgraph generation</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.pos.norm.bw</dt>
<dd>read depth normalized, coverage on positive strand bigWig file</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.pos.norm.bw.out</dt>
<dd>std out for read depth normalized, coverage on positive strand bigWig file</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.rpkm</dt>
<dd>RPKMs calculated from SingleRPKM</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.rpkm.out</dt>
<dd>std out from RPKMs calculated from SingleRPKM</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.splices</dt>
<dd>Old Splice Results</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.splices.out</dt>
<dd>stdout from Old Splice Results</dd>
</dl>
<p>ABANDON ALL HOPE YE WHO ENTER HERE THESE FILES ARE FROM BOKYO FOR GENERATING a-&gt;i editing analysis</p>
<p>polyATrim.adapterTrim.rmRep.sorted.rg_stage1.bcf
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.bcf.bci
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995.bb
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995.bed
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995.csv
fastq.gz.polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995.no100
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995.regions
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.eff10
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.eff5
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.eff5-10
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.noSNP
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.sorted.bam
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.sorted.bam.bai
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.var
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.vcf
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.bam
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.bcf
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.bcf.bci
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf0.995
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf0.995.bed
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf0.995.csv
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf0.995.no100
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf0.995.regions
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.eff10
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.eff5
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.eff5-10
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.noSNP
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.rmdup.bam
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.sorted.bam
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.sorted.bam.bai
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.var
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.vcf
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.bcf
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.bcf.bci
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.eff10
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.eff5
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.eff5-10
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.noSNP
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.sorted.bam
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.sorted.bam.bai
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.var
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.vcf</p>


    </div>
      
  </div>
</div>
<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
    </p>
    <p>
        &copy; Copyright 2014, Olga Botvinnik.<br/>
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.3b1.<br/>
    </p>
  </div>
</footer>
  </body>
</html>