<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>&lt;no title&gt; &#8212; YeoLab-welcome 0.1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head>
  <body role="document">
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <p>The file extension below are generated by RNA-seq pipeline, the prefix for each file is the name fed in by the manifest</p>
<dl class="docutils">
<dt>polyATrim.adapterTrim.fastq.out</dt>
<dd>Std out for cutadapt</dd>
<dt>polyATrim.adapterTrim.metrics</dt>
<dd>cutadapt metrics</dd>
<dt>polyATrim.adapterTrim.rmRep_fastqc.zip</dt>
<dd>zip file for fastqc  post-cutadapt, post, repetitive element removal</dd>
<dt>polyATrim.adapterTrim.rmRep.fastq.out</dt>
<dd>std out for repetitive element removal</dd>
<dt>polyATrim.adapterTrim.rmRep.metrics</dt>
<dd>metrics for number of repetitive elements removed</dd>
<dt>polyATrim.adapterTrim.rmRep.sam</dt>
<dd>sam file mapped to reference genome</dd>
<dt>polyATrim.adapterTrim.rmRep.samLog.final.out</dt>
<dd>STAR specific output</dd>
<dt>polyATrim.adapterTrim.rmRep.samLog.out</dt>
<dd>STAR Specific output</dd>
<dt>polyATrim.adapterTrim.rmRep.samLog.progress.out</dt>
<dd>STAR Specific output</dd>
<dt>polyATrim.adapterTrim.rmRep.samLog.std.out</dt>
<dd>STAR Specific output</dd>
<dt>polyATrim.adapterTrim.rmRep.sam.out</dt>
<dd>STAR mapping std out</dd>
<dt>polyATrim.adapterTrim.rmRep.samSJ.out.tab</dt>
<dd>STAR specific output</dd>
<dt>polyATrim.adapterTrim.rmRep.samUnmapped.out.mate1</dt>
<dd>STAR specific output</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.bai</dt>
<dd>index file for intermediate processing, ignore</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.bam</dt>
<dd>intermediate bam file, ignore</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.bam.out</dt>
<dd>std out for STAR, ignore</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.bai</dt>
<dd>index file for intermediate processing, ignore</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.bam</dt>
<dd>final read, group assigned bam file</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.bam.bai</dt>
<dd>final read, group assigned bam index file</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.bam.bai.out</dt>
<dd>stdout for final read, group assigned bam index file</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.bam.out</dt>
<dd>stdout for final read, group assigned bam file</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.count</dt>
<dd>Counts of reads at in each gene, from CountTags</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.count.out</dt>
<dd>stdout for Counts of reads at in each gene, from CountTags</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.editing.out</dt>
<dd>editing make file out</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.make.out</dt>
<dd>unknown ignore</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.miso.out</dt>
<dd>miso make file out</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.bg.out</dt>
<dd>stdout for genomeCoverageBed, negative strand</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.bw</dt>
<dd>bigWig file showing read coverage along the negative strand, transposed so values are negative</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.bw.out</dt>
<dd>stdout for bw file generation</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.norm.bg.out</dt>
<dd>stdout for bg file generation</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.norm.bw</dt>
<dd>read depth normalized, transposed negative strand coverage bigWig file</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.norm.bw.out</dt>
<dd>stdout for bigwig file generation</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.NRF.metrics</dt>
<dd>metrics for PCR duplication, for reference see: <a class="reference external" href="http://genome.ucsc.edu/ENCODE/qualityMetrics.html">http://genome.ucsc.edu/ENCODE/qualityMetrics.html</a></dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.NRF.metrics.out</dt>
<dd>stdout for NRF calculations</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.pos.bg.out</dt>
<dd>stdout for bedgraph generation</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.pos.bw</dt>
<dd>bigWig file showing coverage along positive strand</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.pos.bw.out</dt>
<dd>stdout for bigwig generation</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.pos.norm.bg.out</dt>
<dd>stdout for bedgraph generation</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.pos.norm.bw</dt>
<dd>read depth normalized, coverage on positive strand bigWig file</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.pos.norm.bw.out</dt>
<dd>std out for read depth normalized, coverage on positive strand bigWig file</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.rpkm</dt>
<dd>RPKMs calculated from SingleRPKM</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.rpkm.out</dt>
<dd>std out from RPKMs calculated from SingleRPKM</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.splices</dt>
<dd>Old Splice Results</dd>
<dt>polyATrim.adapterTrim.rmRep.sorted.rg.splices.out</dt>
<dd>stdout from Old Splice Results</dd>
</dl>
<p>ABANDON ALL HOPE YE WHO ENTER HERE THESE FILES ARE FROM BOKYO FOR GENERATING a-&gt;i editing analysis</p>
<p>polyATrim.adapterTrim.rmRep.sorted.rg_stage1.bcf
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.bcf.bci
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995.bb
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995.bed
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995.csv
fastq.gz.polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995.no100
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995.regions
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.eff10
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.eff5
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.eff5-10
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.noSNP
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.sorted.bam
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.sorted.bam.bai
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.var
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.vcf
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.bam
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.bcf
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.bcf.bci
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf0.995
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf0.995.bed
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf0.995.csv
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf0.995.no100
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf0.995.regions
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.eff10
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.eff5
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.eff5-10
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.noSNP
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.rmdup.bam
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.sorted.bam
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.sorted.bam.bai
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.var
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.vcf
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.bcf
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.bcf.bci
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.eff10
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.eff5
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.eff5-10
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.noSNP
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.sorted.bam
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.sorted.bam.bai
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.var
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.vcf</p>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper"><div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/analyze_rna_seq.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2014, Olga Botvinnik.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.5b1</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.9</a>
      
      |
      <a href="_sources/analyze_rna_seq.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>