<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>TSCC &mdash; YeoLab-welcome 0.1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/basic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/bootswatch-3.2.0/lumen/bootstrap.min.css" type="text/css" />
    <link rel="stylesheet" href="_static/bootstrap-sphinx.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="_static/bootstrap-3.2.0/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="_static/bootstrap-sphinx.js"></script>
    <link rel="top" title="YeoLab-welcome 0.1.0 documentation" href="index.html" />
    <link rel="prev" title="Getting Started" href="getting_started.html" />
<meta charset='utf-8'>
<meta http-equiv='X-UA-Compatible' content='IE=edge,chrome=1'>
<meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1'>
<meta name="apple-mobile-web-app-capable" content="yes">

  </head>
  <body role="document">

  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="index.html">
          Yeolab Onboarding</a>
        <span class="navbar-text navbar-version pull-left"><b>0.1.0</b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
                <li><a href="http://yeolab.ucsd.edu/yeolab/Home.html">YeoLab@UCSD</a></li>
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="index.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul>
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting Started</a><ul>
<li class="toctree-l2"><a class="reference internal" href="getting_started.html#sanford-internet-access">Sanford Internet Access</a></li>
<li class="toctree-l2"><a class="reference internal" href="getting_started.html#printing">Printing</a></li>
<li class="toctree-l2"><a class="reference internal" href="getting_started.html#reserving-rooms-in-scrm-building">Reserving rooms in SCRM building</a></li>
<li class="toctree-l2"><a class="reference internal" href="getting_started.html#access-to-compute-environments">Access to Compute Environments</a></li>
<li class="toctree-l2"><a class="reference internal" href="getting_started.html#online-resources">Online Resources</a></li>
<li class="toctree-l2"><a class="reference internal" href="getting_started.html#internal-resources">Internal Resources</a></li>
</ul>
</li>
</ul>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="">TSCC</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#important-rules">Important rules</a></li>
<li class="toctree-l2"><a class="reference internal" href="#first-steps">First Steps</a></li>
<li class="toctree-l2"><a class="reference internal" href="#installing-and-upgrading-python-packages">Installing and upgrading Python packages</a></li>
<li class="toctree-l2"><a class="reference internal" href="#installing-r-packages-beta">Installing R packages (beta!)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#submitting-and-managing-compute-jobs-on-tscc">Submitting and managing compute jobs on TSCC</a></li>
<li class="toctree-l2"><a class="reference internal" href="#ipython-notebooks-on-tscc">IPython notebooks on TSCC</a></li>
<li class="toctree-l2"><a class="reference internal" href="#random-notes">Random notes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-rna-seq-clip-seq-ribo-seq-etc-qscripts-gatk-queue-pipelines">Running RNA-seq, CLIP-Seq, Ribo-Seq, etc qscripts GATK Queue pipelines</a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">TSCC</a><ul>
<li><a class="reference internal" href="#important-rules">Important rules</a></li>
<li><a class="reference internal" href="#first-steps">First Steps</a><ul>
<li><a class="reference internal" href="#log-on">Log on!</a></li>
<li><a class="reference internal" href="#start-a-screen-session">Start a screen session</a></li>
<li><a class="reference internal" href="#get-gscripts-access-to-software">Get <code class="docutils literal"><span class="pre">gscripts</span></code> access to software</a></li>
<li><a class="reference internal" href="#make-a-virtual-environment-on-tscc">Make a virtual environment on TSCC</a></li>
<li><a class="reference internal" href="#add-the-location-of-genome-to-your-bashrc">Add the location of <code class="docutils literal"><span class="pre">GENOME</span></code> to your <code class="docutils literal"><span class="pre">~/.bashrc</span></code></a></li>
<li><a class="reference internal" href="#organize-your-home-directory">Organize your home directory</a><ul>
<li><a class="reference internal" href="#link-your-scratch-directory-to-your-home">Link your scratch directory to your home</a></li>
<li><a class="reference internal" href="#create-workflow-and-projects-folders">Create workflow and projects folders</a></li>
<li><a class="reference internal" href="#let-us-see-your-stuff">Let us see your stuff</a></li>
</ul>
</li>
<li><a class="reference internal" href="#share-your-dropbox-account-for-easy-figure-syncing">Share your Dropbox account for easy figure syncing</a></li>
</ul>
</li>
<li><a class="reference internal" href="#installing-and-upgrading-python-packages">Installing and upgrading Python packages</a></li>
<li><a class="reference internal" href="#installing-r-packages-beta">Installing R packages (beta!)</a></li>
<li><a class="reference internal" href="#submitting-and-managing-compute-jobs-on-tscc">Submitting and managing compute jobs on TSCC</a><ul>
<li><a class="reference internal" href="#submit-jobs">Submit jobs</a></li>
<li><a class="reference internal" href="#submit-interactive-jobs">Submit interactive jobs</a></li>
<li><a class="reference internal" href="#submit-jobs-to-home-scrm">Submit jobs to <code class="docutils literal"><span class="pre">home-scrm</span></code></a></li>
<li><a class="reference internal" href="#submitting-many-jobs-at-once">Submitting many jobs at once</a></li>
<li><a class="reference internal" href="#check-job-status-aka-why-is-my-job-stuck">Check job status, aka &#8220;why is my job stuck?&#8221;</a></li>
<li><a class="reference internal" href="#check-job-status-of-array-jobs">Check job status of array jobs</a></li>
<li><a class="reference internal" href="#killing-jobs">Killing jobs</a></li>
<li><a class="reference internal" href="#kill-an-array-job">Kill an array job</a></li>
<li><a class="reference internal" href="#kill-all-your-jobs">Kill all your jobs</a></li>
<li><a class="reference internal" href="#which-queue-do-i-submit-to-check-status-of-queues">Which queue do I submit to? (check status of queues)</a></li>
<li><a class="reference internal" href="#show-available-service-units">Show available &#8220;Service Units&#8221;</a></li>
<li><a class="reference internal" href="#show-available-processors">Show available processors</a></li>
<li><a class="reference internal" href="#show-specs-of-all-nodes">Show specs of all nodes</a></li>
</ul>
</li>
<li><a class="reference internal" href="#ipython-notebooks-on-tscc">IPython notebooks on TSCC</a><ul>
<li><a class="reference internal" href="#setup-ipython-notebooks-on-tscc">Setup IPython notebooks on TSCC</a></li>
<li><a class="reference internal" href="#run-ipython-notebooks-on-tscc">Run IPython Notebooks on TSCC</a></li>
</ul>
</li>
<li><a class="reference internal" href="#random-notes">Random notes</a></li>
<li><a class="reference internal" href="#running-rna-seq-clip-seq-ribo-seq-etc-qscripts-gatk-queue-pipelines">Running RNA-seq, CLIP-Seq, Ribo-Seq, etc qscripts GATK Queue pipelines</a><ul>
<li><a class="reference internal" href="#running-gatk-queue-pipeline-scripts">Running GATK Queue pipeline scripts</a></li>
<li><a class="reference internal" href="#pipeline-frequently-asked-questions-faq">Pipeline frequently asked questions (FAQ)</a><ul>
<li><a class="reference internal" href="#how-do-i">How do I ...</a></li>
</ul>
</li>
<li><a class="reference internal" href="#analyze-rna-seq">analyze_rna_seq</a></li>
<li><a class="reference internal" href="#analyze-rna-seq-gently">analyze_rna_seq_gently</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
              
                
  <li>
    <a href="getting_started.html" title="Previous Chapter: Getting Started"><span class="glyphicon glyphicon-chevron-left visible-sm"></span><span class="hidden-sm hidden-tablet">&laquo; Getting Started</span>
    </a>
  </li>
              
            
            
            
            
              <li class="hidden-sm">
<div id="sourcelink">
  <a href="_sources/tscc.txt"
     rel="nofollow">Source</a>
</div></li>
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">
    <div class="col-md-12">
      
  <div class="section" id="tscc">
<h1>TSCC<a class="headerlink" href="#tscc" title="Permalink to this headline">¶</a></h1>
<p><a class="reference external" href="http://rci.ucsd.edu/computing/index.html">TSCC</a> houses our 640-core supercomputer as part of a condo resource sharing
system which allows other researchers (mostly bioinformaticians from the
Ideker, Ren, Zhang labs) to use our portion for their jobs (with an 8 hour
time limit) which allows us to use their portions when we need extra
computing power. We lead the pack in terms of pure crunching power,
with the Ideker lab in close second place with 512 cores. But they have 2x
the RAM per core for jobs that require lots of memory,
so our purchases are complimentary and sharing is encouraged.</p>
<p>The main contacts for questions about TSCC are the <a class="reference external" href="mailto:dryyeo-l&#37;&#52;&#48;googlegroups&#46;com">dry lab</a> and
<a class="reference external" href="mailto:tscc-l&#37;&#52;&#48;mailman&#46;ucsd&#46;edu">TSCC users</a> mailing lists. The main contact for problems with TSCC is <a class="reference external" href="mailto:jhayes&#37;&#52;&#48;sdsc&#46;edu">Jim Hayes</a></p>
<div class="section" id="important-rules">
<h2>Important rules<a class="headerlink" href="#important-rules" title="Permalink to this headline">¶</a></h2>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<ol class="last arabic simple">
<li>All sequencing data is stored in the <code class="docutils literal"><span class="pre">/projects/ps-yeolab/seqdata</span></code> folder</li>
<li>The folder <code class="docutils literal"><span class="pre">seqdata</span></code> is intended as permanent storage and no folders
or files there should ever be deleted</li>
<li>Do not process data in <code class="docutils literal"><span class="pre">seqdata</span></code>. Use the directory structure
described in <a class="reference internal" href="#organize-your-home-directory">Organize your home directory</a> to create a <code class="docutils literal"><span class="pre">scratch</span></code>
folder for all data processing.</li>
</ol>
</div>
</div>
<div class="section" id="first-steps">
<h2>First Steps<a class="headerlink" href="#first-steps" title="Permalink to this headline">¶</a></h2>
<p>Your first login session should include some of the following commands,
which will familiarize you with the cluster, teach you how to do some useful
tasks on the queue, and help you set up a common directory structure shared
by everyone in the lab.</p>
<div class="section" id="log-on">
<h3>Log on!<a class="headerlink" href="#log-on" title="Permalink to this headline">¶</a></h3>
<p>First, log in to TSCC!</p>
<div class="code highlight-python"><div class="highlight"><pre>ssh YOUR_TSCC_USERNAME@tscc-login2.sdsc.edu
</pre></div>
</div>
<p>TSCC has two login nodes, <code class="docutils literal"><span class="pre">login1</span></code> and <code class="docutils literal"><span class="pre">login2</span></code> for load-balancing (i.e.
so if you just log on to <code class="docutils literal"><span class="pre">tscc.sdsc.edu</span></code>, it&#8217;ll choose whichever login
node is less occupied. We&#8217;re logging in to a specific node because then
we&#8217;ll always have our screen session on the same node) This is logging
specifically on to <code class="docutils literal"><span class="pre">login2</span></code>. You can do <code class="docutils literal"><span class="pre">login1</span></code> if you like, as well,
to balance it out :)</p>
</div>
<div class="section" id="start-a-screen-session">
<h3>Start a screen session<a class="headerlink" href="#start-a-screen-session" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="https://kb.iu.edu/d/acuy">Screen</a> is an awesome tool which allows you to have multiple &#8220;tabs&#8221; open in
the same login session, and you can easily transition between screens. Plus,
they&#8217;re persistent, so you can leave something running in a screen session,
log out of TSCC, and it will still be running! Amazing! If you have
suggestions of things to add to this <code class="docutils literal"><span class="pre">.screenrc</span></code>, feel free to make a pull
request on Olga&#8217;s <a class="reference external" href="https://github.com/olgabot/rcfiles">rcfiles</a> github repo.</p>
<p>To get a nice status bar at the bottom of your terminal window, get this
<code class="docutils literal"><span class="pre">.screenrc</span></code> file:</p>
<div class="code highlight-python"><div class="highlight"><pre>cd
wget https://raw.githubusercontent.com/olgabot/rcfiles/master/.screenrc
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The control letter is <code class="docutils literal"><span class="pre">j</span></code>, not <code class="docutils literal"><span class="pre">a</span></code> in the documentation above,
so for example to create a new window, do <code class="docutils literal"><span class="pre">Ctrl-j</span> <span class="pre">c</span></code> and to kill the
current window, do <code class="docutils literal"><span class="pre">Ctrl-j</span> <span class="pre">k</span></code>. Do <code class="docutils literal"><span class="pre">Ctrl-j</span> <span class="pre">j</span></code> to switch between
windows, and <code class="docutils literal"><span class="pre">Ctrl-j</span> <span class="pre">#</span></code>, where # is some window number,
to switch to a numbered tab specifically (e.g. <code class="docutils literal"><span class="pre">Ctrl-j</span> <span class="pre">2</span></code> to switch to
tab number 2.</p>
</div>
<p>This <code class="docutils literal"><span class="pre">.screenrc</span></code> adds a status bar at the bottom of your screen, like this:</p>
<img alt="_images/screen_status.png" src="_images/screen_status.png" />
<p>Now to start a screen session do:</p>
<div class="code highlight-python"><div class="highlight"><pre><span class="n">screen</span>
</pre></div>
</div>
<p>If you&#8217;re re-logging in and you have an old screen session,
do this to &#8220;re-attach&#8221; the screen window.</p>
<div class="code highlight-python"><div class="highlight"><pre><span class="n">screen</span> <span class="o">-</span><span class="n">x</span>
</pre></div>
</div>
<p>Every time you log in to TSCC, you&#8217;ll want to reattach the screens from
before, so the first step I always take when I log in to TSCC is exactly
that, <code class="docutils literal"><span class="pre">screen</span> <span class="pre">-x</span></code>.</p>
</div>
<div class="section" id="get-gscripts-access-to-software">
<h3>Get <code class="docutils literal"><span class="pre">gscripts</span></code> access to software<a class="headerlink" href="#get-gscripts-access-to-software" title="Permalink to this headline">¶</a></h3>
<ol class="arabic simple" start="0">
<li>Before you&#8217;re able to clone the gscripts github repo, you&#8217;ll need to add
your ssh keys on TSCC to your Github account. Follow <a class="reference external" href="https://help.github.com/articles/generating-ssh-keys/">Github&#8217;s instructions
for generating SSH keys</a>.</li>
<li>First, clone the <code class="docutils literal"><span class="pre">gscripts</span></code> github repo to your home directory on TSCC
(this assumes you&#8217;ve already created a github account).</li>
</ol>
<div class="code highlight-python"><div class="highlight"><pre># &lt;on TSCC&gt;
cd
git clone http://github.com/YeoLab/gscripts
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li>Add this line to the <strong>end</strong> of your <code class="docutils literal"><span class="pre">~/.bashrc</span></code> file (using either
<code class="docutils literal"><span class="pre">emacs</span></code> or <code class="docutils literal"><span class="pre">vi</span></code>/<code class="docutils literal"><span class="pre">vim</span></code>, your choice)</li>
</ol>
<div class="code highlight-python"><div class="highlight"><pre>source ~/gscripts/bashrc/tscc_bash_settings_current
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Make sure to add <code class="docutils literal"><span class="pre">source</span> <span class="pre">~/gscripts/bashrc/tscc_bash_settings_current</span></code>
to your <code class="docutils literal"><span class="pre">~/.bashrc</span></code> file so that it always loads up the correct yeolab
environment variables!</p>
</div>
<ol class="arabic" start="3">
<li><dl class="first docutils">
<dt>&#8220;source the <code class="docutils literal"><span class="pre">.bashrc</span></code> file so you load all the convenient environment</dt>
<dd><p class="first last">variables we&#8217;ve created.</p>
</dd>
</dl>
</li>
</ol>
<div class="code highlight-python"><div class="highlight"><pre>source ~/.bashrc
</pre></div>
</div>
</div>
<div class="section" id="make-a-virtual-environment-on-tscc">
<h3>Make a virtual environment on TSCC<a class="headerlink" href="#make-a-virtual-environment-on-tscc" title="Permalink to this headline">¶</a></h3>
<p>On TSCC, the easiest way to create a virtual evironment (aka <code class="docutils literal"><span class="pre">virtualenv</span></code>)
is by making one off of the <code class="docutils literal"><span class="pre">base</span></code> environment, which already has a bunch of
modules that we use all the time (<code class="docutils literal"><span class="pre">numpy</span></code>, <code class="docutils literal"><span class="pre">scipy</span></code>, <code class="docutils literal"><span class="pre">matplotlib</span></code>, <code class="docutils literal"><span class="pre">pandas</span></code>, <code class="docutils literal"><span class="pre">scikit-learn</span></code>, <code class="docutils literal"><span class="pre">ipython</span></code>, the list goes on). Here&#8217;s how you do it:</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>The command <code class="docutils literal"><span class="pre">$USER</span></code> is meant to be literal, meaning you can exactly copy
the below command, and TSCC will create an environment with your username.
If you don&#8217;t believe me, compare the output of:</p>
<div class="code highlight-python"><div class="highlight"><pre>echo USER
</pre></div>
</div>
<p>to the output of:</p>
<div class="code highlight-python"><div class="highlight"><pre>echo $USER
</pre></div>
</div>
<p class="last">The second one should output your TSCC username, because the <code class="docutils literal"><span class="pre">$</span></code> dollar
sign indicates to the shell that you&#8217;re asking for the variable <code class="docutils literal"><span class="pre">$USER</span></code>,
not the literal word &#8220;USER&#8221;.</p>
</div>
<div class="code highlight-python"><div class="highlight"><pre>conda create --clone base --name $USER
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>You can also create an environment from scratch using <code class="docutils literal"><span class="pre">conda</span></code> to install
all the Anaconda Python packages, and then using <code class="docutils literal"><span class="pre">pip</span></code> in the environment
to install the remaining packages, like so:</p>
<div class="code highlight-python"><div class="highlight"><pre>conda create --yes --name ENVIRONMENT_NAME pip numpy scipy cython matplotlib nose six scikit-learn ipython networkx pandas tornado statsmodels setuptools pytest pyzmq jinja2 pyyaml pymongo biopython
conda install --channel https://conda.binstar.org pybedtools
source activate ENVIRONMENT_NAME
pip install seaborn fastcluster gspread brewer2mpl husl semantic_version joblib pybedtools gffutils matplotlib-venn HTSeq misopy
pip install https://github.com/YeoLab/clipper/tarball/master
pip install https://github.com/YeoLab/gscripts/tarball/master
pip install https://github.com/YeoLab/flotilla/tarball/master
</pre></div>
</div>
<p class="last">These commands is how the <code class="docutils literal"><span class="pre">base</span></code> environment was created.</p>
</div>
<p>Then activate your environment with</p>
<div class="code highlight-python"><div class="highlight"><pre>source activate $USER
</pre></div>
</div>
<p>You&#8217;ll probably stay in this environment all the time.</p>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">Make sure to add <code class="docutils literal"><span class="pre">source</span> <span class="pre">activate</span> <span class="pre">$USER</span></code> to your <code class="docutils literal"><span class="pre">~/.bashrc</span></code> file!
Then you will always be in your environment</p>
</div>
<p>If you need to switch to another environment, then exit your environment with:</p>
<div class="code highlight-python"><div class="highlight"><pre>source deactivate
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>Now that you&#8217;ve created your own environment, go to your gscripts folder
and install your own personal gscripts, to make sure it&#8217;s the most updated
version.</p>
<div class="code last highlight-python"><div class="highlight"><pre>cd ~/gscripts
pip install .  # The &quot;.&quot; means install &quot;this,&quot; as in &quot;this folder where I am&quot;
</pre></div>
</div>
</div>
</div>
<div class="section" id="add-the-location-of-genome-to-your-bashrc">
<h3>Add the location of <code class="docutils literal"><span class="pre">GENOME</span></code> to your <code class="docutils literal"><span class="pre">~/.bashrc</span></code><a class="headerlink" href="#add-the-location-of-genome-to-your-bashrc" title="Permalink to this headline">¶</a></h3>
<p>To run the analysis pipeline, you will need to specify where the genomes are
on TSCC, and you can do this by adding this line to your <code class="docutils literal"><span class="pre">~/.bashrc</span></code>:</p>
<div class="code highlight-python"><div class="highlight"><pre>GENOME=/projects/ps-yeolab/genomes
</pre></div>
</div>
</div>
<div class="section" id="organize-your-home-directory">
<h3>Organize your home directory<a class="headerlink" href="#organize-your-home-directory" title="Permalink to this headline">¶</a></h3>
<p>Create an organized <code class="docutils literal"><span class="pre">home</span></code> directory structure following a common
template, so others can find your scripts, workflows,
and even final results/papers!  Do not store actual data in your home
directory as is is limited to 100 GB only.</p>
<div class="section" id="link-your-scratch-directory-to-your-home">
<h4>Link your scratch directory to your home<a class="headerlink" href="#link-your-scratch-directory-to-your-home" title="Permalink to this headline">¶</a></h4>
<p>The &#8220;<code class="docutils literal"><span class="pre">scratch</span></code>&#8221; storage on TSCC is for temporary (after 90 days it gets
purged) storage. It&#8217;s very useful for storing intermediate files,
and outputs from compute jobs because the data there is stored on
solid-state drives (SSDs, currently 300TB) which have incredibly fast
read-write speeds, which is perfect for outputs from alignment algorithms.
It can be annoying to go back and forth between your scratch directory,
so it&#8217;s convenient to have a link to your scratch from home,
which you can make like this:</p>
<div class="code highlight-python"><div class="highlight"><pre>ln -s /oasis/tscc/scratch/$USER $HOME/scratch
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This is virtually unlimited temporary storage space,
designed for heavy I/O.  Aside from common reference files (e.g.
Genomes, GENCODE, etc.) this should be the only space that you can
read/write to from your scripts/workflows! The &#8216;&#8217;&#8216;parallel&#8217;&#8216;&#8217; throughput
of this storage is 100 GB/s (e.g. 10 tasks can each read/write at 10
GB/s at the same time)</p>
</div>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p>Anything saved here is subject to deletion without warning after 3 months
or less of storage. In particular, the large <code class="docutils literal"><span class="pre">.sam</span></code> and <code class="docutils literal"><span class="pre">.bam</span></code> files
can get deleted, even though their <code class="docutils literal"><span class="pre">.done</span></code> files (produced by the
GATK Queue RNA-seq pipeline as a placeholder) will still exist, and they
will seem done to the pipeline. To avoid lost data, here are a few steps:</p>
<ol class="last arabic">
<li><p class="first">Keep your metadata sample/cell counts are in your <code class="docutils literal"><span class="pre">$HOME/projects</span></code> or
<code class="docutils literal"><span class="pre">/projects/ps-yeolab/$USER</span></code> folder, which don&#8217;t get purged
periodically.</p>
</li>
<li><p class="first">Delete <code class="docutils literal"><span class="pre">*.done</span></code> files when re-rerunning a partially eroded pipeline
run.</p>
</li>
<li><p class="first">Use this recursive touch command to &#8220;refresh&#8221; the decay clock on your
files before important meetings and re-analysis steps:</p>
<div class="code highlight-python"><div class="highlight"><pre>cd important_scratch_dir
find . | xargs touch
</pre></div>
</div>
</li>
</ol>
</div>
</div>
<div class="section" id="create-workflow-and-projects-folders">
<h4>Create workflow and projects folders<a class="headerlink" href="#create-workflow-and-projects-folders" title="Permalink to this headline">¶</a></h4>
<p>Create <code class="docutils literal"><span class="pre">~/workflows</span></code> for your personal bash, makefile, queue, and so on,
scripts, before you add them to gscripts, and <code class="docutils literal"><span class="pre">~/projects</span></code> for your
projects to organize figures, notebooks, final results, and even manuscripts.</p>
<div class="code highlight-python"><div class="highlight"><pre>mkdir ~/workflows ~/projects
</pre></div>
</div>
<p>Here&#8217;s an example project directory structure:</p>
<div class="code highlight-python"><div class="highlight"><pre>$ ls -lha /home/gpratt/projects/fox2_iclip/
total 9.5K
drwxr-xr-x  2 gpratt yeo-group  5 Sep 16  2013 .
drwxr-xr-x 40 gpratt yeo-group 40 Nov 24 12:20 ..
lrwxrwxrwx  1 gpratt yeo-group 49 Aug 21  2013 analysis -&gt; /home/gpratt/scratch/projects/fox2_iclip/analysis
lrwxrwxrwx  1 gpratt yeo-group 45 Aug 21  2013 data -&gt; /home/gpratt/scratch/projects/fox2_iclip/data
lrwxrwxrwx  1 gpratt yeo-group 50 Aug 21  2013 scripts -&gt; /home/gpratt/processing_scripts/fox2_iclip/scripts
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Notice that all of these are soft-links to either <code class="docutils literal"><span class="pre">~/scratch</span></code> or some
other processing scripts.</p>
</div>
</div>
<div class="section" id="let-us-see-your-stuff">
<h4>Let us see your stuff<a class="headerlink" href="#let-us-see-your-stuff" title="Permalink to this headline">¶</a></h4>
<p>Make everything readable by other yeo lab members and restrict access from
other users (per HIPAA/HITECH requirements)</p>
<div class="code highlight-python"><div class="highlight"><pre>chmod -R g+r ~/
chmod -R g+r ~/scratch/
chmod -R o-rwx ~/
chmod -R o-rwx ~/scratch/
</pre></div>
</div>
<p>But <code class="docutils literal"><span class="pre">git</span></code> will get mad at you if your ~/.ssh keys private keys are visible
by others, so make them visible to only you via:</p>
<div class="code highlight-python"><div class="highlight"><pre>chmod -R go-rwx ~/.ssh/
</pre></div>
</div>
<p>In the end, your &#8216;&#8217;&#8216;home&#8217;&#8216;&#8217; directory should look something like this:</p>
<div class="code highlight-python"><div class="highlight"><pre>$ ls -l $HOME
lrwxrwxrwx  1 bkakarad yeo-group    29 Jun 24  2013 scratch -&gt; /oasis/tscc/scratch/bkakarad/
drwxr-x---+ 2 bkakarad yeo-group     2 Jun 24  2013 gscripts
drwxr-x---+ 3 bkakarad yeo-group     3 Jun 24  2013 projects
drwxr-x---+ 2 bkakarad yeo-group     2 Jun 24  2013 workflows
</pre></div>
</div>
</div>
</div>
<div class="section" id="share-your-dropbox-account-for-easy-figure-syncing">
<h3>Share your Dropbox account for easy figure syncing<a class="headerlink" href="#share-your-dropbox-account-for-easy-figure-syncing" title="Permalink to this headline">¶</a></h3>
</div>
</div>
<div class="section" id="installing-and-upgrading-python-packages">
<h2>Installing and upgrading Python packages<a class="headerlink" href="#installing-and-upgrading-python-packages" title="Permalink to this headline">¶</a></h2>
<p>To install Python packages first try <code class="docutils literal"><span class="pre">conda</span> <span class="pre">install</span></code>:</p>
<div class="code highlight-python"><div class="highlight"><pre>conda install &lt;package name&gt;
</pre></div>
</div>
<p>If there is no package in conda, then (and ONLY then) try <cite>pip</cite>:</p>
<div class="code highlight-python"><div class="highlight"><pre>pip install &lt;package name&gt;
</pre></div>
</div>
<p>To upgrade packages, do:</p>
<p>(using <code class="docutils literal"><span class="pre">conda</span></code>)</p>
<div class="code highlight-python"><div class="highlight"><pre>conda update &lt;package name&gt;
</pre></div>
</div>
<p>(using <code class="docutils literal"><span class="pre">pip</span></code>)</p>
<div class="code highlight-python"><div class="highlight"><pre>pip install -U &lt;package name&gt;
</pre></div>
</div>
</div>
<div class="section" id="installing-r-packages-beta">
<h2>Installing R packages (beta!)<a class="headerlink" href="#installing-r-packages-beta" title="Permalink to this headline">¶</a></h2>
<p>You can also use <code class="docutils literal"><span class="pre">conda</span></code> to install <code class="docutils literal"><span class="pre">R</span></code> and <code class="docutils literal"><span class="pre">R</span></code> packages. Currently, you
need to reference one of Anaconda&#8217;s developer&#8217;s channel <code class="docutils literal"><span class="pre">asmeurer</span></code> to install
it. Here is the command to install R in your environment. You can see the list
of <a class="reference external" href="https://binstar.org/asmeurer/">R packages he&#8217;s added so far</a>.</p>
<div class="code highlight-python"><div class="highlight"><pre>conda install -c asmeurer r
</pre></div>
</div>
</div>
<div class="section" id="submitting-and-managing-compute-jobs-on-tscc">
<h2>Submitting and managing compute jobs on TSCC<a class="headerlink" href="#submitting-and-managing-compute-jobs-on-tscc" title="Permalink to this headline">¶</a></h2>
<div class="section" id="submit-jobs">
<h3>Submit jobs<a class="headerlink" href="#submit-jobs" title="Permalink to this headline">¶</a></h3>
<p>To submit a script that you wrote, in this case called <code class="docutils literal"><span class="pre">myscript.sh</span></code>,
to TSCC, do:</p>
<div class="code highlight-python"><div class="highlight"><pre>qsub -q home-yeo -l nodes=1:ppn=2 -l walltime=0:30:00 myscript.sh
</pre></div>
</div>
</div>
<div class="section" id="submit-interactive-jobs">
<h3>Submit interactive jobs<a class="headerlink" href="#submit-interactive-jobs" title="Permalink to this headline">¶</a></h3>
<p>To submit interactive jobs, do:</p>
<div class="code highlight-python"><div class="highlight"><pre>qsub -I -q home-yeo -l nodes=1:ppn=2 -l walltime=0:30:00
</pre></div>
</div>
</div>
<div class="section" id="submit-jobs-to-home-scrm">
<h3>Submit jobs to <code class="docutils literal"><span class="pre">home-scrm</span></code><a class="headerlink" href="#submit-jobs-to-home-scrm" title="Permalink to this headline">¶</a></h3>
<p>To submit to the <code class="docutils literal"><span class="pre">home-scrm</span></code> queue, add <code class="docutils literal"><span class="pre">-W</span> <span class="pre">group_list=scrm-group</span></code> to
your <code class="docutils literal"><span class="pre">qsub</span></code> command:</p>
<div class="code highlight-python"><div class="highlight"><pre>qsub -I -l walltime=0:30:00 -q home-scrm -W group_list=scrm-group
</pre></div>
</div>
</div>
<div class="section" id="submitting-many-jobs-at-once">
<h3>Submitting many jobs at once<a class="headerlink" href="#submitting-many-jobs-at-once" title="Permalink to this headline">¶</a></h3>
<p>If you have a bunch of commands you want to run at once,
you can use this script to submit them all at once. In the next example,
<code class="docutils literal"><span class="pre">commands.sh</span></code> is a file has the commands you want on their own line,
i.e. one command per line.</p>
<div class="code highlight-python"><div class="highlight"><pre>java -Xms512m -Xmx512m -jar /home/yeo-lab/software/gatk/dist/Queue.jar \
-S ~/gscripts/qscripts/do_stuff.scala --input commands.sh -run -qsub \
-jobQueue &lt;queue&gt; -jobLimit &lt;n&gt; --ncores &lt;n&gt; --jobname &lt;name&gt; -startFromScratch
</pre></div>
</div>
<p>This runs a scala job that submits sub-jobs to the PBS queue under name you
fill in where &lt;name&gt; now sits as a placeholder.</p>
</div>
<div class="section" id="check-job-status-aka-why-is-my-job-stuck">
<h3>Check job status, aka &#8220;why is my job stuck?&#8221;<a class="headerlink" href="#check-job-status-aka-why-is-my-job-stuck" title="Permalink to this headline">¶</a></h3>
<p>Check the status of your jobs:</p>
<div class="code highlight-python"><div class="highlight"><pre><span class="n">qme</span>
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This will only work if you have followed instructions and have</p>
</div>
<p><code class="docutils literal"><span class="pre">source</span></code>&#8216;d the <code class="docutils literal"><span class="pre">~/gscripts/tscc_bash_settings_current</span></code>  :)</p>
<p><code class="docutils literal"><span class="pre">qme</span></code> outputs,</p>
<div class="code highlight-python"><div class="highlight"><pre>(olga)[obotvinnik@tscc-login2 ~]$ qme

tscc-mgr.sdsc.edu:
                                                                                  Req&#39;d    Req&#39;d       Elap
Job ID                  Username    Queue    Jobname          SessID  NDS   TSK   Memory   Time    S   Time
----------------------- ----------- -------- ---------------- ------ ----- ------ ------ --------- - ---------
2006527.tscc-mgr.local  obotvinnik  home-yeo STDIN             35367     1     16    --   04:00:00 R  02:35:36
2007542.tscc-mgr.local  obotvinnik  home-yeo STDIN              6168     1      1    --   08:00:00 R  00:28:08
2007621.tscc-mgr.local  obotvinnik  home-yeo STDIN               --      1     16    --   04:00:00 Q       --
</pre></div>
</div>
</div>
<div class="section" id="check-job-status-of-array-jobs">
<h3>Check job status of array jobs<a class="headerlink" href="#check-job-status-of-array-jobs" title="Permalink to this headline">¶</a></h3>
<p>Check the status of your array jobs, you need to specify <code class="docutils literal"><span class="pre">-t</span></code> to see the
status of the individual array pieces.</p>
<div class="code highlight-python"><div class="highlight"><pre><span class="n">qstat</span> <span class="o">-</span><span class="n">t</span>
</pre></div>
</div>
</div>
<div class="section" id="killing-jobs">
<h3>Killing jobs<a class="headerlink" href="#killing-jobs" title="Permalink to this headline">¶</a></h3>
<p>If you have a job you want to stop, kill it with <code class="docutils literal"><span class="pre">qdel</span> <span class="pre">JOBID</span></code>, e.g.</p>
<div class="code highlight-python"><div class="highlight"><pre>qdel 2006527
</pre></div>
</div>
</div>
<div class="section" id="kill-an-array-job">
<h3>Kill an array job<a class="headerlink" href="#kill-an-array-job" title="Permalink to this headline">¶</a></h3>
<p>If the job is an array job, you&#8217;ll need to add brackets, like this:</p>
<div class="code highlight-python"><div class="highlight"><pre>qdel 2006527[]
</pre></div>
</div>
</div>
<div class="section" id="kill-all-your-jobs">
<h3>Kill all your jobs<a class="headerlink" href="#kill-all-your-jobs" title="Permalink to this headline">¶</a></h3>
<p>To kill all the jobs that you&#8217;ve submitted, do:</p>
<div class="code highlight-python"><div class="highlight"><pre>qdel $(qselect -u $USER)
</pre></div>
</div>
</div>
<div class="section" id="which-queue-do-i-submit-to-check-status-of-queues">
<h3>Which queue do I submit to? (check status of queues)<a class="headerlink" href="#which-queue-do-i-submit-to-check-status-of-queues" title="Permalink to this headline">¶</a></h3>
<p>Check the status of the queue (so you know which queues to NOT submit to!)</p>
<div class="code highlight-python"><div class="highlight"><pre><span class="n">qstat</span> <span class="o">-</span><span class="n">q</span>
</pre></div>
</div>
<p>Example output is,</p>
<div class="code highlight-python"><div class="highlight"><pre>(olga)[obotvinnik@tscc-login2 ~]$ qstat -q

server: tscc-mgr.local

Queue            Memory CPU Time Walltime Node  Run Que Lm  State
---------------- ------ -------- -------- ----  --- --- --  -----
home-dkeres        --      --       --      --    2   0 --   E R
home-komunjer      --      --       --      --    0   0 --   E R
home-ong           --      --       --      --    2   0 --   E R
home-tg            --      --       --      --    0   0 --   E R
home-yeo           --      --       --      --    3   1 --   E R
home-visres        --      --       --      --    0   0 --   E R
home-mccammon      --      --       --      --   15  29 --   E R
home-scrm          --      --       --      --    1   0 --   E R
hotel              --      --    168:00:0   --  232  26 --   E R
home-k4zhang       --      --       --      --    0   0 --   E R
home-kkey          --      --       --      --    0   0 --   E R
home-kyang         --      --       --      --    2   1 --   E R
home-jsebat        --      --       --      --    1   0 --   E R
pdafm              --      --    72:00:00   --    1   0 --   E R
condo              --      --    08:00:00   --   18   6 --   E R
gpu-hotel          --      --    336:00:0   --    0   0 --   E R
glean              --      --       --      --   24  75 --   E R
gpu-condo          --      --    08:00:00   --   16  36 --   E R
home-fpaesani      --      --       --      --    4   2 --   E R
home-builder       --      --       --      --    0   0 --   E R
home               --      --       --      --    0   0 --   E R
home-mgilson       --      --       --      --    0   4 --   E R
home-eallen        --      --       --      --    0   0 --   E R
                                               ----- -----
                                                 321   180
</pre></div>
</div>
<p>So right now is not a good time to submit to the <code class="docutils literal"><span class="pre">hotel</span></code> queue,
since it has a bunch of both running and queued jobs!</p>
</div>
<div class="section" id="show-available-service-units">
<h3>Show available &#8220;Service Units&#8221;<a class="headerlink" href="#show-available-service-units" title="Permalink to this headline">¶</a></h3>
<p>List the available Service Units (1 SU = 1 core*hour) ... for a quick ego
boost. Also note that our supercomputer is separated in two: yeo-group and
scrm-group, but the total balance is 5.29 million SU, just enough secure us
the top honors :-)</p>
<div class="code highlight-python"><div class="highlight"><pre>gbalance | sort -nrk 3 | head

Id Name                 Amount  Reserved Balance CreditLimit Available
-- -------------------- ------- -------- ------- ----------- ---------
19 tideker-group        5211035    27922 5183113           0   5183113
82 yeo-group            3262925        0 3262925           0   3262925
81 scrm-group           2039328        0 2039328           0   2039328
14 mgilson-group         663095   208000  455095           0    455095
73 nanosprings-ucm       650000        0  650000           0    650000
17 kkey-group            635056     7104  627952           0    627952
16 k4zhang-group         534430        0  534430           0    534430
</pre></div>
</div>
<p>List the available TORQUE queues, for a quick boost in motivation!</p>
<div class="code highlight-python"><div class="highlight"><pre>qstat -q

Queue            Memory CPU Time Walltime Node  Run Que Lm  State
---------------- ------ -------- -------- ----  --- --- --  -----
home-tideker       --      --       --       16   1   0 --   E R
home-visres        --      --       --        1   0   0 --   E R
hotel              --      --    72:00:00   --   25  18 --   E R
home-k4zhang       --      --       --        4  21   0 --   E R
home-kkey          --      --       --        5   0   0 --   E R
pdafm              --      --    72:00:00   --    0   0 --   E R
condo              --      --    08:00:00   --    0   0 --   E R
glean              --      --       --      --    0   0 --   E R
home-builder       --      --       --        8   0   0 --   E R
home               --      --       --      --    0   0 --   E R
home-ewyeo         --      --       --       15   0   0 --   E R
home-mgilson       --      --       --        8   0   0 --   E R
                                           ----- -----
                                              47    18
</pre></div>
</div>
</div>
<div class="section" id="show-available-processors">
<h3>Show available processors<a class="headerlink" href="#show-available-processors" title="Permalink to this headline">¶</a></h3>
<p>To show available processors, do</p>
<div class="code highlight-python"><div class="highlight"><pre><span class="n">showbf</span>
</pre></div>
</div>
</div>
<div class="section" id="show-specs-of-all-nodes">
<h3>Show specs of all nodes<a class="headerlink" href="#show-specs-of-all-nodes" title="Permalink to this headline">¶</a></h3>
<div class="code highlight-python"><div class="highlight"><pre><span class="n">pbsnodes</span> <span class="o">-</span><span class="n">a</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="ipython-notebooks-on-tscc">
<h2>IPython notebooks on TSCC<a class="headerlink" href="#ipython-notebooks-on-tscc" title="Permalink to this headline">¶</a></h2>
<div class="section" id="setup-ipython-notebooks-on-tscc">
<h3>Setup IPython notebooks on TSCC<a class="headerlink" href="#setup-ipython-notebooks-on-tscc" title="Permalink to this headline">¶</a></h3>
<ol class="arabic simple">
<li>To set up IPython notebooks on TSCC, you will want to add some <code class="docutils literal"><span class="pre">alias</span></code>
variables to your <code class="docutils literal"><span class="pre">~/.bashrc</span></code>. First, on your personal computer,
you will want to set up
<a class="reference external" href="http://www.linuxproblem.org/art_9.html">passwordless ssh</a> from your laptop to TSCC. On my laptop,
I have this alias in my <cite>~/.bashrc</cite> file:</li>
</ol>
<div class="code highlight-python"><div class="highlight"><pre>  IPYNB_PORT=[some number above 1024]
  alias tscc=&#39;ssh obotvinnik@tscc-login2.sdsc.edu&#39;

This way, I can just type ``tscc`` and log onto ``tscc-login2``
**specifically**. It is important for IPython notebooks that you always log
on to the same node. You can use ``tscc-login1`` instead, too,
this is just what I have set up. Just replace my login name
(&quot;``obotvinnik``&quot;) with yours.
</pre></div>
</div>
<ol class="arabic" start="2">
<li><p class="first">Next, type <code class="docutils literal"><span class="pre">tscc</span></code> and log on to the server.</p>
</li>
<li><p class="first">On TSCC, add these lines to your <code class="docutils literal"><span class="pre">~/.bashrc</span></code> file.</p>
<div class="code highlight-python"><div class="highlight"><pre>IPYNB_PORT=[same number as the above IPYNB_PORT]
alias ipynb=&quot;ipython notebook --no-browser --port $IPYNB_PORT --matplotlib inline &amp;&quot;
alias sshtscc=&quot;ssh -NR $IPYNB_PORT:localhost:$IPYNB_PORT tscc-login2 &amp;&quot;
</pre></div>
</div>
<p>Notice that in <code class="docutils literal"><span class="pre">sshtscc</span></code>, I use the same port as I logged in to,
<cite>tscc-login2</cite>. The ampersands &#8220;<cite>&amp;</cite>&#8221; at the end of the lines tell the computer
to run these processes in the background, which is super useful.</p>
</li>
<li><p class="first">Set up passwordless ssh between the compute nodes and TSCC with:</p>
</li>
</ol>
<div class="code highlight-python"><div class="highlight"><pre>cat .ssh/id_rsa.pub | ssh tscc-login2 &#39;cat &gt;&gt; .ssh/authorized_keys&#39;
</pre></div>
</div>
<ol class="arabic" start="5">
<li><p class="first">Back on your home laptop, edit your <cite>~/.bash_profile</cite> on macs,
<cite>~/.bashrc</cite> for other unix machines to add the lines:</p>
<div class="code highlight-python"><div class="highlight"><pre>IPYNB_PORT=[the same number as you chose above]
tunneltscc=&quot;ssh -NL $IPYNB_PORT:localhost:$IPYNB_PORT YOUR_TSCC_USERNAME@tscc-login2.sdsc.edu &amp;&quot;
</pre></div>
</div>
<p>Make sure to replace &#8220;<code class="docutils literal"><span class="pre">YOUR_TSCC_USERNAME</span></code>&#8221; with your TSCC login :) It is
also important that these are double-quotes and not single-quotes.</p>
</li>
</ol>
</div>
<div class="section" id="run-ipython-notebooks-on-tscc">
<h3>Run IPython Notebooks on TSCC<a class="headerlink" href="#run-ipython-notebooks-on-tscc" title="Permalink to this headline">¶</a></h3>
<p>Now that you have everything configured, you can run IPython notebooks on TSCC!
Here are the steps to follow.</p>
<p>1. Log on to TSCC
4. Now that you have those set up, start up a <code class="docutils literal"><span class="pre">screen</span></code> session,</p>
<blockquote>
<div>which allows you to have something running continuously,
without being logged in.</div></blockquote>
<div class="code highlight-python"><div class="highlight"><pre> screen -x

.. note::
    If this gives you an error saying &quot;There is no screen to be attached&quot;
    then you need to run plain old ``screen`` (no ``-x``) first.

    If this gives you an error saying you need to pick one session, make
    life easier for yourself and pick one to kill all the windows in,
    (using ``Ctrl-j K`` if you&#39;re using the ``.screenrc`` that I recommended
    earlier, otherwise the default is ``Ctrl-a K``). Once you&#39;ve killed all
    screen sessions except for one, you can run ``screen -x`` with abandon,
    and it will connect you to the only one you have open.
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li>In this <code class="docutils literal"><span class="pre">screen</span></code> session, now request an interactive job, e.g.:</li>
</ol>
<div class="code highlight-python"><div class="highlight"><pre>qsub -I -l walltime=2:00:00 -q home-yeo -l nodes=1:ppn=2
</pre></div>
</div>
<ol class="arabic simple" start="3">
<li>Wait for the job to start.</li>
<li>Run your TSCC-specific aliases on the compute node:</li>
</ol>
<div class="code highlight-python"><div class="highlight"><pre><span class="n">ipynb</span>
<span class="n">sshtscc</span>
</pre></div>
</div>
<ol class="arabic simple" start="4">
<li><strong>Back on your laptop</strong>, now run your tunneling command:</li>
</ol>
<div class="code highlight-python"><div class="highlight"><pre><span class="n">tunneltscc</span>
</pre></div>
</div>
<ol class="arabic simple" start="5">
<li>Open up <code class="docutils literal"><span class="pre">http://localhost:[YOUR</span> <span class="pre">IPYNB</span> <span class="pre">PORT]</span></code> on your browser.</li>
</ol>
</div>
</div>
<div class="section" id="random-notes">
<h2>Random notes<a class="headerlink" href="#random-notes" title="Permalink to this headline">¶</a></h2>
<p>Software goes in <code class="docutils literal"><span class="pre">/projects/ps-yeolab/software</span></code></p>
<p>Make sure to recursively set group read/write permissions to the software
directory so others can use and update the common software, using:</p>
<div class="code highlight-python"><div class="highlight"><pre>chmod ug+rw /projects/ps-yeolab/software
</pre></div>
</div>
<p>If your&#8217;e installing something from source and using <code class="docutils literal"><span class="pre">./configure</span></code>
and <code class="docutils literal"><span class="pre">make</span></code> and all that, then always set the flag
<code class="docutils literal"><span class="pre">--prefix=/projects/ps-yeolab/software</span></code> when you run <code class="docutils literal"><span class="pre">./configure</span></code></p>
<div class="code highlight-python"><div class="highlight"><pre>./configure --prefix /projects/ps-yeolab/software
</pre></div>
</div>
<p>When possible install bins to <code class="docutils literal"><span class="pre">/projects/ps-yeolab/software/bin</span></code></p>
</div>
<div class="section" id="running-rna-seq-clip-seq-ribo-seq-etc-qscripts-gatk-queue-pipelines">
<h2>Running RNA-seq, CLIP-Seq, Ribo-Seq, etc qscripts GATK Queue pipelines<a class="headerlink" href="#running-rna-seq-clip-seq-ribo-seq-etc-qscripts-gatk-queue-pipelines" title="Permalink to this headline">¶</a></h2>
<p>We use the Broad Institute&#8217;s Genome Analysis Toolkit (<a class="reference external" href="https://www.broadinstitute.org/gatk/">GATK</a>) <a class="reference external" href="http://gatkforums.broadinstitute.org/discussion/1306/overview-of-queue">Queue</a> software
to run our pipelines. This software solves a lot of problems for us, such as
dealing with multiple-stage pipelines that have cross-dependencies (e.g. you
can&#8217;t calculate splicing until you&#8217;ve mapped the reads, and you can&#8217;t map
the reads until after you&#8217;ve removed adapters and repetitive genomic regions
from them), and properly scheduling jobs so that one person&#8217;s analysis doesn&#8217;t
completely take over the compute cluster.</p>
<p>Gabe has created a bunch of helpful template scripts for GATK Queue in his
folder <code class="docutils literal"><span class="pre">/home/gpratt/templates</span></code>:</p>
<div class="code highlight-python"><div class="highlight"><pre>$ ls -lh /home/gpratt/templates
total 26K
-rwxr-xr-x 1 gpratt yeo-group 660 May  7  2014 bacode_split.sh
-rwxr-xr-x 1 gpratt yeo-group 554 May  7  2014 bacode_split.sh~
-rwxr-xr-x 1 gpratt yeo-group 524 Sep 18 00:08 #clipseq.sh#
-rwxr-xr-x 1 gpratt yeo-group 524 Jul 12  2014 clipseq.sh
-rwxr-xr-x 1 gpratt yeo-group 516 Mar 26  2014 clipseq.sh~
-rwxr-xr-x 1 gpratt yeo-group 473 Aug 21 18:47 riboseq.sh
-rwxr-xr-x 1 gpratt yeo-group 528 Aug 21 18:46 riboseq.sh~
-rwxr-xr-x 1 gpratt yeo-group 530 Sep  5 17:29 rnaseq.sh
-rwxr-xr-x 1 gpratt yeo-group 527 Mar 26  2014 rnaseq.sh~
</pre></div>
</div>
<p>Each Queue job requires a manifest file with a list of all files to process,
and the genome to process them on.</p>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p>All further instructions depend on you having followed the directions in
<a class="reference internal" href="#create-workflow-and-projects-folders">Create workflow and projects folders</a>, where for this particular project,
you&#8217;ve created these folders:</p>
<div class="code highlight-python"><div class="highlight"><pre>~/projects/PROJECT_NAME
~/processing_scripts/PROJECT_NAME/scripts
~/scratch/PROJECT_NAME/data
~/scratch/PROJECT_NAME/analysis
</pre></div>
</div>
<p>And that you&#8217;ve linked the scratch and home directories correctly. For
example, here&#8217;s how you can create the project directory structure for a
project called <code class="docutils literal"><span class="pre">singlecell_pnms</span></code>:</p>
<div class="code last highlight-python"><div class="highlight"><pre>NAME=singlecell_pnms
mkdir -p ~/projects/$NAME ~/scratch/$NAME ~/scratch/$NAME/data ~/scratch/$NAME/analysis ~/processing_scripts/$NAME/scripts
ln -s ~/scratch/$NAME/data ~/projects/$NAME/data
ln -s ~/scratch/$NAME/analysis ~/projects/$NAME/analysis
ln -s ~/processing_scripts/$NAME/scripts ~/projects/$NAME/scripts
</pre></div>
</div>
</div>
<p>Here&#8217;s an example queue script for single-end, not strand-specific RNA-seq,
from the file <code class="docutils literal"><span class="pre">singlecell_pnms_se_v3.sh</span></code>:</p>
<div class="code highlight-python"><div class="highlight"><pre>#!/bin/bash

NAME=singlecell_pnms_se
VERSION=v3
DIR=singlecell_pnms
java -Xms512m -Xmx512m -jar /projects/ps-yeolab/software/gatk/dist/Queue.jar -S $HOME/gscripts/qscripts/analyze_rna_seq.scala --input ${NAME}_${VERSION}.txt --adapter TCGTATGCCGTCTTCTGCTTG --adapter ATCTCGTATGCCGTCTTCTGCTTG --adapter CGACAGGTTCAGAGTTCTACAGTCCGACGATC --adapter GATCGGAAGAGCACACGTCTGAACTCCAGTCAC -qsub -jobQueue home-yeo -runDir ~/projects/${DIR}/analysis/${NAME}_${VERSION}  -log ${NAME}_${VERSION}.log --location ${NAME}  --strict -keepIntermediates --not_stranded -single_end -run
</pre></div>
</div>
<p>Notice that the &#8220;<code class="docutils literal"><span class="pre">--input</span></code>&#8221; is the file <code class="docutils literal"><span class="pre">${NAME}_${VERSION}.txt</span></code>, which
translates to <code class="docutils literal"><span class="pre">singlecell_pnms_se_v3.txt</span></code> in this case, since
<code class="docutils literal"><span class="pre">NAME=singlecell_pnms_se</span></code> and <code class="docutils literal"><span class="pre">VERSION=v3</span></code> are defined at the beginning of
the file. This file is the &#8220;manifest&#8221; of the sequencing run. In the case of
single-end reads, this is a file where each line has,
<code class="docutils literal"><span class="pre">/path/to/read1.fastq.gz\tspecies\n</span></code>,
where <code class="docutils literal"><span class="pre">\t</span></code> indicates a tab (using the <code class="docutils literal"><span class="pre">&lt;TAB&gt;</span></code> character), and <code class="docutils literal"><span class="pre">\n</span></code>
indicates a new line, created by <code class="docutils literal"><span class="pre">&lt;ENTER&gt;</span></code>. Here is the first
10 lines of <code class="docutils literal"><span class="pre">singlecell_pnms_se_v3.txt</span></code> (obtained via
<code class="docutils literal"><span class="pre">head</span> <span class="pre">singlecell_pnms_se_v3.txt</span></code>):</p>
<div class="code highlight-python"><div class="highlight"><pre>/home/obotvinnik/projects/singlecell_pnms/data/CVN_01_R1.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/CVN_02_R1.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/CVN_03_R1.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/CVN_04_R1.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/CVN_05_R1.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/CVN_06_R1.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/CVN_07_R1.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/CVN_08_R1.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/CVN_09_R1.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/CVN_10_R1.fastq.gz       hg19
</pre></div>
</div>
<p>For paired-end, not strand-specific RNA-seq, here&#8217;s the script of the file
<code class="docutils literal"><span class="pre">singlecell_pnms_pe_v3.sh</span></code></p>
<div class="code highlight-python"><div class="highlight"><pre>#!/bin/bash

NAME=singlecell_pnms_pe
VERSION=v3
DIR=singlecell_pnms
java -Xms512m -Xmx512m -jar /projects/ps-yeolab/software/gatk/dist/Queue.jar -S $HOME/gscripts/qscripts/analyze_rna_seq.scala --input ${NAME}_${VERSION}.txt --adapter TCGTATGCCGTCTTCTGCTTG --adapter ATCTCGTATGCCGTCTTCTGCTTG --adapter CGACAGGTTCAGAGTTCTACAGTCCGACGATC --adapter GATCGGAAGAGCACACGTCTGAACTCCAGTCAC -qsub -jobQueue home-yeo -runDir ~/projects/${DIR}/analysis/${NAME}_${VERSION}  -log ${NAME}_${VERSION}.log --location ${NAME}  --strict -keepIntermediates --not_stranded -run
</pre></div>
</div>
<p>Notice that the &#8220;<code class="docutils literal"><span class="pre">--input</span></code>&#8221; is the file <code class="docutils literal"><span class="pre">${NAME}_${VERSION}.txt</span></code>, which
translates to <code class="docutils literal"><span class="pre">singlecell_pnms_pe_v3.txt</span></code> in this case, since
<code class="docutils literal"><span class="pre">NAME=singlecell_pnms_pe</span></code> and <code class="docutils literal"><span class="pre">VERSION=v2</span></code> are defined at the beginning of
the file. This file is the &#8220;manifest&#8221; of the sequencing run. In the case of
single-end reads, this is a file where each line has:
<code class="docutils literal"><span class="pre">read1.fastq.gz;read2.fastq.gz\tspecies\n</span></code>,
where <code class="docutils literal"><span class="pre">\t</span></code> indicates a tab (using the <code class="docutils literal"><span class="pre">&lt;TAB&gt;</span></code> character), and <code class="docutils literal"><span class="pre">\n</span></code>
indicates a new line, created by <code class="docutils literal"><span class="pre">&lt;ENTER&gt;</span></code>. Here is the first
10 lines of <code class="docutils literal"><span class="pre">singlecell_pnms_pe_v3.txt</span></code> (obtained via
<code class="docutils literal"><span class="pre">head</span> <span class="pre">singlecell_pnms_pe_v3.txt</span></code>):</p>
<div class="code highlight-python"><div class="highlight"><pre>/home/obotvinnik/projects/singlecell_pnms/data/M1_01_R1.fastq.gz;/home/obotvinnik/projects/singlecell_pnms/data/M1_01_R2.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/M1_02_R1.fastq.gz;/home/obotvinnik/projects/singlecell_pnms/data/M1_02_R2.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/M1_03_R1.fastq.gz;/home/obotvinnik/projects/singlecell_pnms/data/M1_03_R2.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/M1_04_R1.fastq.gz;/home/obotvinnik/projects/singlecell_pnms/data/M1_04_R2.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/M1_05_R1.fastq.gz;/home/obotvinnik/projects/singlecell_pnms/data/M1_05_R2.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/M1_06_R1.fastq.gz;/home/obotvinnik/projects/singlecell_pnms/data/M1_06_R2.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/M1_07_R1.fastq.gz;/home/obotvinnik/projects/singlecell_pnms/data/M1_07_R2.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/M1_08_R1.fastq.gz;/home/obotvinnik/projects/singlecell_pnms/data/M1_08_R2.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/M1_09_R1.fastq.gz;/home/obotvinnik/projects/singlecell_pnms/data/M1_09_R2.fastq.gz       hg19
/home/obotvinnik/projects/singlecell_pnms/data/M1_10_R1.fastq.gz;/home/obotvinnik/projects/singlecell_pnms/data/M1_10_R2.fastq.gz       hg19
</pre></div>
</div>
<p>For this project, I had a mix of both paired-end and single-end reads, so
that&#8217;s why <code class="docutils literal"><span class="pre">DIR</span></code> is the same for both the <code class="docutils literal"><span class="pre">singlecell_pnms_se_v3.sh</span></code> and
<code class="docutils literal"><span class="pre">singlecell_pnms_pe_v3.sh</span></code> scripts, but <code class="docutils literal"><span class="pre">NAME</span></code> was different - then they&#8217;re
saved in different folders.</p>
<div class="section" id="running-gatk-queue-pipeline-scripts">
<h3>Running GATK Queue pipeline scripts<a class="headerlink" href="#running-gatk-queue-pipeline-scripts" title="Permalink to this headline">¶</a></h3>
<p>Now that you&#8217;ve created manifest file called <code class="docutils literal"><span class="pre">${NAME}_${VERSION}.txt</span></code> and
<code class="docutils literal"><span class="pre">${NAME}_${VERSION}.sh</span></code>, you are almost ready to run the pipeline.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>You should be using <code class="docutils literal"><span class="pre">screen</span></code> quite often now. You&#8217;ll want to run your
pipeline in a <code class="docutils literal"><span class="pre">screen</span></code> session, because then even when you log out of
TSCC, the pipeline will still be running.</p>
<p>If you&#8217;ve already run <code class="docutils literal"><span class="pre">screen</span></code>, reattach the session with:</p>
<div class="code highlight-python"><div class="highlight"><pre><span class="n">screen</span> <span class="o">-</span><span class="n">x</span>
</pre></div>
</div>
<p>If that gives you the error: <code class="docutils literal"><span class="pre">There</span> <span class="pre">is</span> <span class="pre">no</span> <span class="pre">screen</span> <span class="pre">to</span> <span class="pre">be</span> <span class="pre">attached.</span></code>, then
you haven&#8217;t run <code class="docutils literal"><span class="pre">screen</span></code> before, and you can start a session with:</p>
<div class="code last highlight-python"><div class="highlight"><pre><span class="n">screen</span>
</pre></div>
</div>
</div>
<p>These scripts take quite a bit of memory to compile, so to be nice to everyone,
log into a compute node by requesting an interactive job on TSCC. Also your
script may just run out of memory and fail if you&#8217;re not a compute node, so
that is even more incentive to log into a compute node!</p>
<p>This command will create an interactive job for 40 hours, on the <code class="docutils literal"><span class="pre">home-scrm</span></code>
queue, and with 1 node and 1 processor (you don&#8217;t need more than that for the
script, and the script will submit jobs that request more nodes/processors for
compute-intensive stuff like STAR or Sailfish). If you have a lot of samples,
you may need more time, but try just 40 hours first.</p>
<p>So here&#8217;s what you do:</p>
<div class="code highlight-python"><div class="highlight"><pre>qsub -I -l walltime=40:00:00 -q home-scrm
# Wait for the job to be ready. This may take a while
cd ~/projects/$NAME/scripts
sh ${NAME}_${VERSION}.sh
</pre></div>
</div>
<p>For example, for the <code class="docutils literal"><span class="pre">singlecell_pnms</span></code> project from before, I would do:</p>
<div class="code highlight-python"><div class="highlight"><pre>qsub -I -l walltime=40:00:00 -q home-scrm
# Waited for job to get scheduled/be ready ....
cd ~/projects/singlecell_pnms/scripts
sh singlecell_pnms_se_v3.sh
</pre></div>
</div>
<p>This outputs:</p>
<div class="code highlight-python"><div class="highlight"><pre>INFO  12:24:42,840 QScriptManager - Compiling 1 QScript
INFO  12:24:55,100 QScriptManager - Compilation complete
INFO  12:24:55,359 HelpFormatter - ----------------------------------------------------------------------
INFO  12:24:55,359 HelpFormatter - Queue v2.3-1095-gdb26a3f, Compiled 2015/01/26 15:22:32
INFO  12:24:55,359 HelpFormatter - Copyright (c) 2012 The Broad Institute
INFO  12:24:55,359 HelpFormatter - For support and documentation go to http://www.broadinstitute.org/gatk
INFO  12:24:55,360 HelpFormatter - Program Args: -S /home/obotvinnik/gscripts/qscripts/analyze_rna_seq.scala --input singlecell_pnms_se_v3.txt --adapter TCGTATGCCGTCTTCTGCTTG --adapter ATCTCGTATGCCGTCTTCTGCTTG --adapter CGACAGGTTCAGAGTTCTACAGTCCGACGATC --adapter GATCGGAAGAGCACACGTCTGAACTCCAGTCAC -qsub -jobQueue home-yeo -runDir /home/obotvinnik/projects/singlecell_pnms/analysis/singlecell_pnms_se_v3 -log singlecell_pnms_se_v3.log --location singlecell_pnms_se --strict -keepIntermediates --not_stranded -single_end -run
INFO  12:24:55,360 HelpFormatter - Date/Time: 2015/01/27 12:24:55
INFO  12:24:55,360 HelpFormatter - ----------------------------------------------------------------------
INFO  12:24:55,361 HelpFormatter - ----------------------------------------------------------------------
INFO  12:24:55,370 QCommandLine - Scripting AnalyzeRNASeq
INFO  12:24:58,436 QCommandLine - Added 773 functions
INFO  12:24:58,438 QGraph - Generating graph.
INFO  12:24:58,664 QGraph - Running jobs.
... more output ...
</pre></div>
</div>
</div>
<div class="section" id="pipeline-frequently-asked-questions-faq">
<h3>Pipeline frequently asked questions (FAQ)<a class="headerlink" href="#pipeline-frequently-asked-questions-faq" title="Permalink to this headline">¶</a></h3>
<div class="section" id="how-do-i">
<h4>How do I ...<a class="headerlink" href="#how-do-i" title="Permalink to this headline">¶</a></h4>
<dl class="docutils">
<dt>... deal with multiple species? Do I have to create different manifest files?</dt>
<dd>Fortunately, no! You can create a single manifest file.</dd>
</dl>
<p>Looking at <code class="docutils literal"><span class="pre">/home/gpratt/projects/msi2/scripts</span></code>, we see the file
<code class="docutils literal"><span class="pre">msi2_v2.txt</span></code>, which has the contents:</p>
<div class="code highlight-python"><div class="highlight"><pre>/home/gpratt/projects/msi2/data/msi2/MSI2_ACAGTG_ACAGTG_L008_R1.fastq.gz        hg19
/home/gpratt/projects/msi2/data/msi2/MSI2_CAGATC_CAGATC_L008_R1.fastq.gz        mm9
/home/gpratt/projects/msi2/data/msi2/MSI2_GCCAAT_GCCAAT_L008_R1.fastq.gz        mm9
/home/gpratt/projects/msi2/data/msi2/MSI2_TAGCTT_TAGCTT_L008_R1.fastq.gz        hg19
/home/gpratt/projects/msi2/data/msi2/MSI2_TGACCA_TGACCA_L008_R1.fastq.gz        hg19
/home/gpratt/projects/msi2/data/msi2/MSI2_TTAGGC_TTAGGC_L008_R1.fastq.gz        hg19
</pre></div>
</div>
<p>So you can reference multiple genomes in a single manifest file!</p>
<dl class="docutils">
<dt>... deal with both single-end and paired-end reads in one project? Do I need to create separate manifest files?</dt>
<dd>Yes, unfortunately. :( Check out the <code class="docutils literal"><span class="pre">singlecell_pnms</span></code> project above as
an example.</dd>
<dt>... see the documentation for a queue script?</dt>
<dd>This command will show documentation for <code class="docutils literal"><span class="pre">analyze_rna_seq.scala</span></code>. For
further documentation, see the <a class="reference external" href="http://gatkforums.broadinstitute.org/discussion/1306/overview-of-queue">GATK Queue website</a>.</dd>
</dl>
<div class="code highlight-python"><div class="highlight"><pre>java -Xms512m -Xmx512m -jar /projects/ps-yeolab/software/gatk/dist/Queue.jar -S ~/gscripts/qscripts/analyze_rna_seq.scala
</pre></div>
</div>
</div>
</div>
<div class="section" id="analyze-rna-seq">
<h3>analyze_rna_seq<a class="headerlink" href="#analyze-rna-seq" title="Permalink to this headline">¶</a></h3>
<p>The queue script <code class="docutils literal"><span class="pre">analyze_rna_seq.scala</span></code> runs or generates:</p>
<ol class="arabic simple">
<li><a class="reference external" href="http://www.broadinstitute.org/cancer/cga/rna-seqc">RNA-SeQC</a></li>
<li>cutadapt</li>
<li>miso</li>
<li>OldSplice</li>
<li>Sailfish</li>
<li>A-&gt;I editing predictions</li>
<li>bigWig files</li>
<li>Counts of reads mapping to repetitive elements</li>
<li>Estimates of PCR Duplication</li>
</ol>
<p>Detailed description of <a class="reference external" href="analyze_rna_seq">analyze_rna_seq.scala</a> outputs.</p>
</div>
<div class="section" id="analyze-rna-seq-gently">
<h3>analyze_rna_seq_gently<a class="headerlink" href="#analyze-rna-seq-gently" title="Permalink to this headline">¶</a></h3>
<p>The queue script <code class="docutils literal"><span class="pre">analyze_rna_seq_gently.scala</span></code> runs:</p>
<ol class="arabic simple">
<li><a class="reference external" href="http://www.broadinstitute.org/cancer/cga/rna-seqc">RNA-SeQC</a></li>
<li>...</li>
</ol>
</div>
</div>
</div>


    </div>
      
  </div>
</div>
<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
    </p>
    <p>
        &copy; Copyright 2014, Olga Botvinnik.<br/>
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.3.1.<br/>
    </p>
  </div>
</footer>
  </body>
</html>