The file extension below are generated by RNA-seq pipeline, the \
prefix for each file is the name fed in by the manifest

polyATrim.adapterTrim.fastq.out
    Std out for cutadapt

polyATrim.adapterTrim.metrics
    cutadapt metrics

polyATrim.adapterTrim.rmRep_fastqc.zip
    zip file for fastqc  post-cutadapt, post, repetitive element removal

polyATrim.adapterTrim.rmRep.fastq.out
    std out for repetitive element removal

polyATrim.adapterTrim.rmRep.metrics
    metrics for number of repetitive elements removed

polyATrim.adapterTrim.rmRep.sam
    sam file mapped to reference genome

polyATrim.adapterTrim.rmRep.samLog.final.out
    STAR specific output

polyATrim.adapterTrim.rmRep.samLog.out
    STAR Specific output

polyATrim.adapterTrim.rmRep.samLog.progress.out
    STAR Specific output

polyATrim.adapterTrim.rmRep.samLog.std.out
    STAR Specific output

polyATrim.adapterTrim.rmRep.sam.out
    STAR mapping std out

polyATrim.adapterTrim.rmRep.samSJ.out.tab
    STAR specific output

polyATrim.adapterTrim.rmRep.samUnmapped.out.mate1
    STAR specific output

polyATrim.adapterTrim.rmRep.sorted.bai
    index file for intermediate processing, ignore

polyATrim.adapterTrim.rmRep.sorted.bam
    intermediate bam file, ignore

polyATrim.adapterTrim.rmRep.sorted.bam.out
    std out for STAR, ignore

polyATrim.adapterTrim.rmRep.sorted.rg.bai
    index file for intermediate processing, ignore

polyATrim.adapterTrim.rmRep.sorted.rg.bam
    final read, group assigned bam file

polyATrim.adapterTrim.rmRep.sorted.rg.bam.bai
    final read, group assigned bam index file

polyATrim.adapterTrim.rmRep.sorted.rg.bam.bai.out
    stdout for final read, group assigned bam index file

polyATrim.adapterTrim.rmRep.sorted.rg.bam.out
    stdout for final read, group assigned bam file

polyATrim.adapterTrim.rmRep.sorted.rg.count
    Counts of reads at in each gene, from CountTags

polyATrim.adapterTrim.rmRep.sorted.rg.count.out
    stdout for Counts of reads at in each gene, from CountTags

polyATrim.adapterTrim.rmRep.sorted.rg.editing.out
    editing make file out

polyATrim.adapterTrim.rmRep.sorted.rg.make.out
    unknown ignore

polyATrim.adapterTrim.rmRep.sorted.rg.miso.out
    miso make file out

polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.bg.out
    stdout for genomeCoverageBed, negative strand

polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.bw
    bigWig file showing read coverage along the negative strand, transposed so values are negative

polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.bw.out
    stdout for bw file generation

polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.norm.bg.out
    stdout for bg file generation

polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.norm.bw
    read depth normalized, transposed negative strand coverage bigWig file

polyATrim.adapterTrim.rmRep.sorted.rg.neg.t.norm.bw.out
    stdout for bigwig file generation

polyATrim.adapterTrim.rmRep.sorted.rg.NRF.metrics
    metrics for PCR duplication, for reference see: http://genome.ucsc.edu/ENCODE/qualityMetrics.html

polyATrim.adapterTrim.rmRep.sorted.rg.NRF.metrics.out
    stdout for NRF calculations

polyATrim.adapterTrim.rmRep.sorted.rg.pos.bg.out
    stdout for bedgraph generation

polyATrim.adapterTrim.rmRep.sorted.rg.pos.bw
    bigWig file showing coverage along positive strand

polyATrim.adapterTrim.rmRep.sorted.rg.pos.bw.out
    stdout for bigwig generation

polyATrim.adapterTrim.rmRep.sorted.rg.pos.norm.bg.out
    stdout for bedgraph generation

polyATrim.adapterTrim.rmRep.sorted.rg.pos.norm.bw
    read depth normalized, coverage on positive strand bigWig file

polyATrim.adapterTrim.rmRep.sorted.rg.pos.norm.bw.out
    std out for read depth normalized, coverage on positive strand bigWig file

polyATrim.adapterTrim.rmRep.sorted.rg.rpkm
    RPKMs calculated from SingleRPKM

polyATrim.adapterTrim.rmRep.sorted.rg.rpkm.out
    std out from RPKMs calculated from SingleRPKM

polyATrim.adapterTrim.rmRep.sorted.rg.splices
    Old Splice Results

polyATrim.adapterTrim.rmRep.sorted.rg.splices.out
    stdout from Old Splice Results


ABANDON ALL HOPE YE WHO ENTER HERE THESE FILES ARE FROM BOKYO FOR GENERATING a->i editing analysis
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.bcf
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.bcf.bci
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995.bb
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995.bed
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995.csv
fastq.gz.polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995.no100
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.conf0.995.regions
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.eff10
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.eff5
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.eff5-10
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.noSNP
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.sorted.bam
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.sorted.bam.bai
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.var
polyATrim.adapterTrim.rmRep.sorted.rg_stage1.vcf
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.bam
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.bcf
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.bcf.bci
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf0.995
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf0.995.bed
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf0.995.csv
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf0.995.no100
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.conf0.995.regions
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.eff10
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.eff5
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.eff5-10
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.noSNP
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.rmdup.bam
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.sorted.bam
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.sorted.bam.bai
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.var
polyATrim.adapterTrim.rmRep.sorted.rg_stage2.vcf
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.bcf
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.bcf.bci
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.eff10
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.eff5
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.eff5-10
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.noSNP
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.sorted.bam
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.sorted.bam.bai
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.var
polyATrim.adapterTrim.rmRep.sorted.rg_stage3.vcf

